guides/document_lifecycle.md

```diff
@@ -0,0 +1,269 @@
+# Document Lifecycle
+
+This guide covers OpenSearch Ruby Client API actions for Document Lifecycle. You'll learn how to create, read, update, and delete documents in your OpenSearch cluster. Whether you're new to OpenSearch or an experienced user, this guide provides the information you need to manage your document lifecycle effectively.
+
+## Setup
+
+Assuming you have OpenSearch running locally on port 9200, you can create a client instance
+with the following code:
+
+```rust
+let url = Url::parse("http://localhost:9200")?;
+let conn_pool = SingleNodeConnectionPool::new(url);
+let transport = TransportBuilder::new(conn_pool).disable_proxy().build()?;
+let client = OpenSearch::new(transport);
```

Comment:
Let's make this client construction match the guide here https://github.com/opensearch-project/opensearch-rs/blob/main/guides/index_template.md

```suggestion
let url = Url::parse("https://localhost:9200")?;
let credentials = Credentials::Basic("admin".into(), "admin".into());
let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
    .cert_validation(CertificateValidation::None)
    .auth(credentials)
    .build()?;
let client = OpenSearch::new(transport);
```